subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/wifi_mgmt_ext.h>
+#include <net/wifi_credentials.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+
+#include "includes.h"
+#include "common.h"
+#include "common/defs.h"
+#include "wpa_supplicant/config.h"
+#include "wpa_supplicant_i.h"
+#include "driver_i.h"
+
+extern struct k_sem wpa_supplicant_ready_sem;
+extern struct k_mutex wpa_supplicant_mutex;
+extern struct wpa_global *global;
```

Comment:
This looks super confusing, why would the library access `wpa_supplicant` internal variables? I assume it's related to thread safety given the type of variables, but it really should be covered by some helper API on the supplicant side...

Reply:
I didn't change the supplicant side because that approach was previously criticized.
I think the whole command would better be implemented in `supp_api.c`, but having it here in NCS is something like an easy way out. I meddle with internal things because there is just no proper interface to `wpa_supplicant`. I'm currently looking at `wpa_supplicant_ctrl_iface`. Maybe I can use that for the purpose (although it has this awful string-based interface).

Reply:
Just finished looking at `wpa_supplicant_ctrl_iface` and since this still needs internal headers, it doesn't help whatsoever.

Reply:
> I didn't change the supplicant side because that approach was previously criticized.

I agree with @rlubos. I would like to understand this better.